src/shared/components/SubmissionPage/FilestackFilePicker/index.jsx

```diff
@@ -100,8 +100,32 @@ class FilestackFilePicker extends React.Component {
       setFileName,
       setUploadProgress,
       uploadProgress,
+      communitiesList,
+      groups,
     } = this.props;
 
+    let pickupSources = [
+      'local_file_system',
+      'googledrive',
+      'dropbox',
+      'onedrive',
+      'github',
+      'url',
+    ];
+
+    if (communitiesList.data && communitiesList.data.length > 0) {
+      const topGearCommunity = _.find(communitiesList.data, { mainSubdomain: 'topgear' });
+      if (topGearCommunity) {
+        _.forOwn(groups, (value, key) => {
```

Comment:
1. You must check that `communitiesList` is up-to-date (based on the timestamp in Redux store), and if it is not render the standard loading indicator instead of the filepicker. Otherwise, user will see all upload options if he opens filepicker before the list is loaded.

2. For optimal performance the logic should be reversed. First, check whether the challenge belongs to any group (most of challenges do not). If a challenge does not belong to any group, then we do not need to load / wait for `communitiesList` at all, we just render filepicker with all options on.

Reply:
@birdofpreyru implemented in https://github.com/topcoder-platform/community-app/pull/1286